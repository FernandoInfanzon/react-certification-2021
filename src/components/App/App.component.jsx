import React, { useState, useContext } from 'react';
import { BrowserRouter as Router, Switch, Route } from 'react-router-dom';

import Home from '../../pages/Home';
import LoginPage from '../../pages/Login';
import NotFound from '../../pages/NotFound';
import Video from '../../pages/Video';
import VideoFav from '../../pages/VideoFav';
import Layout from '../Layout';
import { Favorites } from '../../pages/private/Favorites';
import Private from '../Private';
import ScrollToTop from '../ScrollToTop';

import ThemeContext from '../../context/ThemeContext';

import youtubeApi from '../../api/youtube';

function App() {
  const { favoritoSelected, favoritosList } = useContext(ThemeContext);

  const [videoList, setVideoList] = useState({
    videosMetaInfo: [],
    selectedVideoId: null,
    selectedVideoDescription: null,
    selectedVideoTitle: null,
  });

  const onSearch = async (keyword) => {
    const response = await youtubeApi.get('/search', {
      params: {
        q: keyword,
      },
    });
    setVideoList({
      videosMetaInfo: response.data.items,
      selectedVideoId: response.data.items[0].id.videoId,
    });
    console.log(videoList.videosMetaInfo);
  };

  return (
    <Router>
      <Layout onSearch={onSearch}>
        <Switch>
          <Route exact path="/">
            <Home videoList={videoList} setVideoList={setVideoList} />
          </Route>
          <Route exact path="/login" component={LoginPage} />
          <Private exact path="/favorites">
            <Favorites />
          </Private>
          <Route exact path="/video/:id">
            <Video videoList={videoList} setVideoList={setVideoList} />
            <ScrollToTop />
          </Route>
          <Private exact path="/video_favorito/:id">
            <VideoFav favoritoSelected={favoritoSelected} favoritosList={favoritosList} />
          </Private>
          <Route path="*">
            <NotFound />
          </Route>
        </Switch>
      </Layout>
    </Router>
  );
}

export default App;
