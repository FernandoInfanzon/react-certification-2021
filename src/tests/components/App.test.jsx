import React from 'react';
import renderer from 'react-test-renderer';
import App from '../../components/App';

describe('App Test', () => {
  test('App Component', () => {
    const component = renderer.create(<App />);
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
