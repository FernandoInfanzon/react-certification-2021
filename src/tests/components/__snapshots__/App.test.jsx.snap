// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App Test App Component 1`] = `
<div
  className="bg-light"
>
  <nav
    className="navbar navbar-expand-lg navbar-light bg-techno"
    id="arriba"
  >
    <div
      className="container-fluid"
    >
      <a
        className="navbar-brand text-uppercase text-dark"
        href="/"
        onClick={[Function]}
      >
        YouTube client app
      </a>
      <button
        aria-controls="navbarText"
        aria-expanded="false"
        aria-label="Toggle navigation"
        className="navbar-toggler"
        data-bs-target="#navbarText"
        data-bs-toggle="collapse"
        type="button"
      >
        <span
          className="navbar-toggler-icon"
        />
      </button>
      <div
        className="collapse navbar-collapse"
        id="navbarText"
      >
        <ul
          className="navbar-nav me-auto mb-2 mb-lg-0 ms-lg-5"
        >
          <li
            className="sc-gtsrHT cnolFt nav-item"
          >
            <a
              aria-current="page"
              className="nav-link active text-dark"
              href="/"
              onClick={[Function]}
            >
              Home
            </a>
          </li>
          <li
            className="sc-gtsrHT cnolFt"
          >
            <div
              className="form-check form-switch d-flex align-items-center"
            >
              <div
                className="pt-2"
              >
                <input
                  className="form-check-input"
                  id="darkMode"
                  name="darkMode"
                  onClick={[Function]}
                  type="checkbox"
                />
                <label
                  className="form-check-label"
                  htmlFor="darkMode"
                >
                  Dark Mode
                </label>
              </div>
            </div>
          </li>
        </ul>
        <span
          className="navbar-text"
        >
          <form
            className="d-flex flex-row "
            onSubmit={[Function]}
          >
            <input
              aria-label="Search"
              className="form-control me-2"
              name="word"
              onChange={[Function]}
              placeholder="Search"
              type="text"
              value="mexico"
            />
            <button
              className="sc-fujyAs wgCwF btn btn-outline-success text-dark"
              type="submit"
            >
              Search
            </button>
            <a
              className="ps-4"
              href="/login"
              onClick={[Function]}
            >
              <svg
                className="bi bi-person-circle"
                fill="black"
                height="30"
                viewBox="0 0 16 16"
                width="50"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M11 6a3 3 0 1 1-6 0 3 3 0 0 1 6 0z"
                />
                <path
                  d="M0 8a8 8 0 1 1 16 0A8 8 0 0 1 0 8zm8-7a7 7 0 0 0-5.468 11.37C3.242 11.226 4.805 10 8 10s4.757 1.225 5.468 2.37A7 7 0 0 0 8 1z"
                  fillRule="evenodd"
                />
              </svg>
            </a>
          </form>
        </span>
      </div>
    </div>
  </nav>
  <main
    className="container pb-5"
  >
    <h1
      className="sc-bdnxRM bZjIND"
      role="heading"
    >
      Welcome to the Challenge!
    </h1>
    <div
      className="container"
    >
      <div
        className="row justify-content-center g-3"
      />
    </div>
  </main>
  <footer
    className="footer mt-auto py-3 bg-dark bg-gradient"
  >
    <div
      className="container text-center"
    >
      <p
        className="sc-kEqXSa EsNga text-muted text-center"
      >
        Luis F. Infanz√≥n M.
      </p>
      <p
        className="sc-iqAclL izIWgQ text-muted text-center"
      >
        Wizeline Academy - 2021 React Bootcamp
      </p>
      <p
        className="sc-iqAclL izIWgQ text-muted text-center"
      >
        Mini Challenge 5
      </p>
    </div>
  </footer>
</div>
`;
